src/storage/v3/vertex.hpp

```diff
@@ -13,24 +13,28 @@
 
 #include <limits>
 #include <tuple>
+#include <type_traits>
 #include <vector>
 
 #include "storage/v3/delta.hpp"
 #include "storage/v3/edge_ref.hpp"
 #include "storage/v3/id_types.hpp"
+#include "storage/v3/key_store.hpp"
 #include "storage/v3/property_store.hpp"
+#include "storage/v3/property_value.hpp"
 #include "utils/spin_lock.hpp"
 
 namespace memgraph::storage::v3 {
 
 struct Vertex {
-  Vertex(Gid gid, Delta *delta) : gid(gid), deleted(false), delta(delta) {
+  Vertex(Gid gid, Delta *delta) : keys{{PropertyValue{gid.AsInt()}}}, deleted(false), delta(delta) {
     MG_ASSERT(delta == nullptr || delta->action == Delta::Action::DELETE_OBJECT,
               "Vertex must be created with an initial DELETE_OBJECT delta!");
   }
 
-  Gid gid;
+  Gid Gid() const { return Gid::FromInt(keys.GetKey(0).ValueInt()); }
 
+  KeyStore keys{{}};
```

Comment:
Why not add a default constructor?

Reply:
Also every vertex will have KeyStore and every `KeyStore` has a `PropertyStore`, isn't `PropertyStore` a global object and local for every vertex, or am I missing something?

Reply:
What do you mean `PropertyStore` being global and local? The `Vertex` class has a `PropertyStore` which basically is the list of the properties in compressed form?

Reply:
@kostasrim Because I don't want to have a default constructor, because the `KeyStore` is meant to be an immutable type. I will remove this though, as it was just a temporary hack.

Reply:
After discussion this with Jure, the important things to note:
1. `PropertyStore` is not a global object at any level, it is somewhat similar to flatmap (property id -> property value) that tries to minimize the used memory. So both `KeyStore` and `Vertex` will have its own `PropertyStore`. Th former one will contain the immutable primary key, the latter one will contain all of the other properties.